extractor/extractor.py

```diff
@@ -0,0 +1,223 @@
+    # check if content is text
+    isText, text = is_text(node)
+    if isText == 1:
+        _content = contents._Content()
+        _content.contentType = 'Text'
+        _content.text = text
+        return 1, _content
+
+    # check if content is image
+    isImage, image = is_image(node)
+    if isImage == 1:
+        _content = contents._Content()
+        _content.contentType = 'Image'
+        _content.image = image
+        return 1, _content
+
+    # check if content is video
+    isVideo, video = is_video(node)
+    if isVideo == 1:
+        _content = contents._Content()
+        _content.contentType = 'Video'
+        _content.video = video
+        return 1, _content
+
+    # check if content is quote
+    # check if content is tweet
+    # check if content is instagram post
+    # check if content is pinterest pin
+    # check if content is youtube video
+
+    return 0, {}
+
+
+def iterate_over_dom(node, contents_list):
+    isContent, _content = is_content(node)
+
+    # Add to the final list if a valid content is extracted
+    if isContent == 1:
+        contents_list.add_content(_content)
+
+    if node is not None and node.name is not None and isContent == 0:
+        for child in node.contents:
+            iterate_over_dom(child, contents_list)
+
+
+def empty_text(_str):
+    return _str == '\n' or _str == ' ' or len(_str) == 0
+
+
+def not_important_text(node):
+    heading = {'h1','h2','h3','h4','h5','h6'}
+    if 0 < len(node.get_text()) < 50 and node.name not in heading:
+        return True
+    else:
+        return False
+
+
+def is_text(node):
+    all_text_tags = {'p', 'span', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'code'}
+
+    if type(node) == bs4.element.Tag:
+        text_data = node.get_text().strip()
+        if (node.name in all_text_tags) and not empty_text(text_data) and not not_important_text(node):
+            text = contents.Text()
+            text.type = node.name
+            text.text_string = text_data
+            return 1, text
+    elif type(node) == bs4.element.NavigableString and not empty_text(node) and len(node) > 50:
+        node = node.strip()
+        text = contents.Text()
+        text.type = ''
+        text.text_string = node
+        return 1, text
+
+    return 0, {}
+
+
+def get_image_size(url):
+    global req_session
+    try:
+        img = PIL.Image.open(req_session.get(url).text)
+        return img.size
+    except ValueError:
+        pass
+
+    return 0, 0
+
+
+def get_image_attributes(node,image):
+    image.img_url = node['src']
+    if node.has_attr('title'):
+        image.img_title = node['title']
+    h, w = get_image_size(image.img_url)
+    image.img_height = h
+    image.img_width = w
+    if image.img_url.endswith('.gif'):
+        image.is_gif = True
+
+
+def is_image(node):
+    if node.name == 'img' and node.has_attr('src'):
+        image = contents.Image()
+        get_image_attributes(node, image)
+        return 1, image
+
+    if node.name == 'figure':
```

Comment:
When does 'figure' case occur?

Reply:
Sometimes instead of using <img> directly it's used as a child of <figure> with <figcaption> as sibling of <img>.